### ISSUE TITLE
.NET 5 RC 1

### ISSUE THREAD

**Issue by rbhanda** [dotnet/core#5200]
[Release Notes](https://github.com/dotnet/core/blob/master/release-notes/5.0/preview/5.0.0-rc.1.md)

Please report any issues you find with .NET 5, either responding to this issue, creating a new issue or creating a new issue in one of the following repos:

* [dotnet/aspnetcore](https://github.com/dotnet/aspnetcore/issues) - for ASP.NET Core questions
* [dotnet/efcore](https://github.com/dotnet/efcore/issues) - for Entity Framework Core questions
* [dotnet/sdk](https://github.com/dotnet/sdk/issues) - for CLI tools and questions
* [dotnet/runtime](https://github.com/dotnet/runtime/issues) - for runtime, API, or installer issues and questions
* [dotnet/winforms](https://github.com/dotnet/winforms/issues) - for WinForms issues
* [dotnet/wpf](https://github.com/dotnet/wpf/issues) - for WPF issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

### Known Issues
If there are any issues with the September 2020 release we will track them here and check issues off as they're resolved. See the linked issues for details on progress and resolution details.

- [x] dotnet-runtime-5.0.0-rc.1.20451.14-win-x86.exe & dotnet-sdk-5.0.100-rc.1.20452.10-win-x64.exe are not signed. - This issue was resolved by updating links on [releases.json](https://github.com/dotnet/core/blob/master/release-notes/5.0/releases.json).  The [release notes](https://github.com/dotnet/core/blob/master/release-notes/5.0/preview/5.0.0-rc.1.md) and [dot.net](https://dotnet.microsoft.com/download/dotnet/5.0) were updated with correct links

- [x] Microsoft.DotNet.Common.ItemTemplates.5.0.0-rc.1.20431.3. nupkg, Microsoft.DotNet.Common.ProjectTemplates.1.x.5.0.0-rc.1.20431.3.nupkg, Microsoft.DotNet.Common.ProjectTemplates.2.0.5.0.0-rc.1.20431.3.nupkg are not pushed to [nuget.org](https://www.nuget.org/)

- [x] [releases.json](https://github.com/dotnet/core/blob/master/release-notes/5.0/releases.json) support phase is currently set to "preview". This needs an update to reflect support phase as "rc".

- [x] Discrepancy in VS support for .NET 5 RC 1 in [releases.json](https://github.com/dotnet/core/blob/master/release-notes/5.0/releases.json). This was resolved by https://github.com/dotnet/core/pull/5203

**Comment by SalvadorSarpi** [comment]
same as #5202.
i've had no problem with preview versions in the past, but rc1 sdk wont install on my windows dev machine (windows 10 pro, preview 8 installed and working). 

![image](https://user-images.githubusercontent.com/12118077/93118776-2a655500-f697-11ea-9346-86c1c5eebea3.png)

```
[20F4:2148][2020-09-14T14:27:05]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{aab15ccd-d49f-4845-9563-2dab405b3188}, resume: Active, restart initiated: No, disable resume: No
[0AD0:43B4][2020-09-14T14:27:05]i336: Acquiring container: WixAttachedContainer, copy from: C:\Users\xx\Desktop\dotnet-sdk-5.0.100-rc.1.20452.10-win-x64.exe
[0AD0:43B4][2020-09-14T14:27:05]i000: Setting string variable 'WixBundleLastUsedSource' to value 'C:\Users\xx\Desktop\'
[0AD0:3FC4][2020-09-14T14:27:05]e000: Error 0x80070570: Failed to extract all files from container, erf: 1:4:0
[0AD0:43B4][2020-09-14T14:27:05]e000: Error 0x80070570: Failed to begin and wait for operation.
[0AD0:43B4][2020-09-14T14:27:05]e000: Error 0x80070570: Failed to extract payload: a3 from container: WixAttachedContainer
[0AD0:43B4][2020-09-14T14:27:05]e312: Failed to extract payloads from container: WixAttachedContainer to working path: C:\Users\xx\AppData\Local\Temp\{FF1A4D3F-1710-42CA-9F35-B9861FBA8638}\8C09F345E398B4710CE74EE6AAFC4D9F91CB9A11, error: 0x80070570.
[0AD0:41D4][2020-09-14T14:27:05]e000: Error 0x80070570: Cache thread exited unexpectedly.
[20F4:2148][2020-09-14T14:27:05]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{aab15ccd-d49f-4845-9563-2dab405b3188}, resume: None, restart: None, disable resume: No
[20F4:2148][2020-09-14T14:27:05]i330: Removed bundle dependency provider: {aab15ccd-d49f-4845-9563-2dab405b3188}
[20F4:2148][2020-09-14T14:27:05]i352: Removing cached bundle: {aab15ccd-d49f-4845-9563-2dab405b3188}, from path: C:\ProgramData\Package Cache\{aab15ccd-d49f-4845-9563-2dab405b3188}\
[20F4:2148][2020-09-14T14:27:05]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{aab15ccd-d49f-4845-9563-2dab405b3188}, resume: None, restart initiated: No, disable resume: No
[0AD0:41D4][2020-09-14T14:27:06]i399: Apply complete, result: 0x80070570, restart: None, ba requested restart:  No

```

**Comment by rbhanda** [comment]
Hi @SalvadorSarpi we are in the process of republishing the files. I will update the issue once it is resolved.

**Comment by rbhanda** [comment]
Hi,
The files are updated on https://github.com/dotnet/core/blob/master/release-notes/5.0/preview/5.0.0-rc.1.md
We are in the process of updating them on [dot.net](https://dotnet.microsoft.com/download/dotnet/5.0)

**Comment by SalvadorSarpi** [comment]
working fine now!

**Comment by helios2052** [comment]
I noticed that in both my upgraded app and in the sample RC1 blazor-wasm hosted template (w/auth), it will startup and run ok the first time, but throws an exception on the next and subsequent startups w/debugging.

It seems like possibly this error only happens if you stop the app/browser by pressing "stop debugging" in visual studio and then re-launch. If you close the browser directly, visual studio then stops debugging as usual and you can restart the app with no issues. I have to go into task manager and clear out any stuck ".NET HOST" processes and then relaunch each time.


![error](https://user-images.githubusercontent.com/291534/93139318-84254980-f6ae-11ea-8f16-05227e4496dd.JPG)

<pre>
Unhandled exception. System.IO.IOException: Failed to bind to address http://127.0.0.1:9300: address already in use.
 ---> Microsoft.AspNetCore.Connections.AddressInUseException: Only one usage of each socket address (protocol/network address/port) is normally permitted.
 ---> System.Net.Sockets.SocketException (10048): Only one usage of each socket address (protocol/network address/port) is normally permitted.
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Bind(EndPoint localEP)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.<Bind>g__BindSocket|13_0(<>c__DisplayClass13_0& )
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.<Bind>g__BindSocket|13_0(<>c__DisplayClass13_0& )
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportFactory.BindAsync(EndPoint endpoint, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, ConnectionDelegate connectionDelegate, EndpointConfig endpointConfig)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass29_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.LocalhostListenOptions.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IEnumerable`1 listenOptions, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.WebHost.StartAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token, String startupMessage)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token, String startupMessage)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host)
   at Microsoft.WebAssembly.Diagnostics.Program.Main(String[] args)

fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.InvalidOperationException: No output has been recevied from the application.
         at Microsoft.AspNetCore.Builder.DebugProxyLauncher.LaunchAndGetUrl(IServiceProvider serviceProvider, String devToolsHost)
         at Microsoft.AspNetCore.Builder.WebAssemblyNetDebugProxyAppBuilderExtensions.<>c.<<UseWebAssemblyDebugging>b__0_1>d.MoveNext()
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.MigrationsEndPointMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
         at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
<pre/>

**Comment by captainsafia** [comment]
@helios2052 Thanks for reporting this. It's a known issue with .NET 5 RC1. We migrated to a new debugging server that was set to use static port 9300. This issue will be fixed in RC2 ([ref](https://github.com/dotnet/runtime/pull/41715)).

In the meantime, I recommend ensuring that no process is running on port 9300 before starting debugging and killing it if so. .NET 5 RC2 will listen on a dynamic port.

Thanks in advance for your patience as we improve the experience here!

**Comment by rbhanda** [comment]
https://github.com/dotnet/core/issues/5200#issuecomment-692361736

**Comment by suleymanoz** [comment]
Hi,
We are using protobuf-net.grpc with http (not https). In preview 8, gRPC with http was working fine. After rc1 upgrade http does not work, but https works.

**Comment by RebelionTheGrey** [comment]
> Hi,
> We are using protobuf-net.grpc with http (not https). In preview 8, gRPC with http was working fine. After rc1 upgrade http does not work, but https works.

Have the same problem. Haven't checked https yet.

**Comment by suleymanoz** [comment]
> Hi,
> We are using protobuf-net.grpc with http (not https). In preview 8, gRPC with http was working fine. After rc1 upgrade http does not work, but https works.

Upgrading preview version for Grpc packages solved the problem.

**Comment by daemogar** [comment]
When using the **WebAssemblyPrerendered** in a _Host page for rendering a Blazor app, Renders as expected on the server, but fails with the below error when trying to activate the WebAssembly client-side. I have tried using the **ServerPrerendered** and it fails even on the server now, and this app was working before the update to .NET 5 RC1. I following the instructions at the link (Blog Posted on gitter.im by @danroth27) below for _Upgrade an existing project_ and _Blazor WebAssembly prerendering_.

So I started a new project and only made the minimum necessary changes to do a **WebAssemblyPrerendered** project that is a DotNetCore Hosted server with the _Host page.

I get the same error in either both my existing project or the new project.

```
blazor.webassembly.js:1 crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Cannot read property 'insertBefore' of null
      TypeError: Cannot read property 'insertBefore' of null
          at s (https://localhost:5001/_framework/blazor.webassembly.js:1:8901)
          at Object.t.createAndInsertLogicalContainer (https://localhost:5001/_framework/blazor.webassembly.js:1:9936)
          at e.insertComponent (https://localhost:5001/_framework/blazor.webassembly.js:1:32392)
          at e.insertFrame (https://localhost:5001/_framework/blazor.webassembly.js:1:31007)
          at e.applyEdits (https://localhost:5001/_framework/blazor.webassembly.js:1:29142)
          at e.updateComponent (https://localhost:5001/_framework/blazor.webassembly.js:1:28458)
          at Object.t.renderBatch (https://localhost:5001/_framework/blazor.webassembly.js:1:11924)
          at Object.window.Blazor._internal.renderBatch (https://localhost:5001/_framework/blazor.webassembly.js:1:57967)
          at Object.w [as invokeJSFromDotNet] (https://localhost:5001/_framework/blazor.webassembly.js:1:60490)
          at _mono_wasm_invoke_js_blazor (https://localhost:5001/_framework/dotnet.5.0.0-rc.1.20451.14.js:1:183003)
Microsoft.JSInterop.JSException: Cannot read property 'insertBefore' of null
TypeError: Cannot read property 'insertBefore' of null
    at s (https://localhost:5001/_framework/blazor.webassembly.js:1:8901)
    at Object.t.createAndInsertLogicalContainer (https://localhost:5001/_framework/blazor.webassembly.js:1:9936)
    at e.insertComponent (https://localhost:5001/_framework/blazor.webassembly.js:1:32392)
    at e.insertFrame (https://localhost:5001/_framework/blazor.webassembly.js:1:31007)
    at e.applyEdits (https://localhost:5001/_framework/blazor.webassembly.js:1:29142)
    at e.updateComponent (https://localhost:5001/_framework/blazor.webassembly.js:1:28458)
    at Object.t.renderBatch (https://localhost:5001/_framework/blazor.webassembly.js:1:11924)
    at Object.window.Blazor._internal.renderBatch (https://localhost:5001/_framework/blazor.webassembly.js:1:57967)
    at Object.w [as invokeJSFromDotNet] (https://localhost:5001/_framework/blazor.webassembly.js:1:60490)
    at _mono_wasm_invoke_js_blazor (https://localhost:5001/_framework/dotnet.5.0.0-rc.1.20451.14.js:1:183003)
   at Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime.Microsoft.JSInterop.IJSUnmarshalledRuntime.InvokeUnmarshalled[Int32,RenderBatch,Object,Object](String identifier, Int32 arg0, RenderBatch arg1, Object arg2)
   at Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime.Microsoft.JSInterop.IJSUnmarshalledRuntime.InvokeUnmarshalled[Int32,RenderBatch,Object](String identifier, Int32 arg0, RenderBatch arg1)
   at Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer.UpdateDisplayAsync(RenderBatch& batch)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.ProcessRenderQueue()
f.printErr @ blazor.webassembly.js:1
```

Blog Post by @danroth27 
https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-5-release-candidate-1/

**Comment by MaxMommersteeg** [comment]
I recommend following SO answer for killing the process listening to the specific port (9300):
https://stackoverflow.com/a/39633428/1661209

**Comment by crozone** [comment]
The latest .NET 5 host does not appear to be respecting `global.json`, at least on my Windows 10 install. Anyone else run into this?

https://github.com/dotnet/roslyn/issues/43782

**Comment by reinux** [comment]
.NET SDK 5.0.100-rc.1.20452.10 install had issues; now it refuses to uninstall -- or rather, it says immediately after pressing Uninstall that it's done uninstalling, other than let me reinstall, which also happens instantly.

Specifically, when I try to create a new F# project of any sort, `dotnet new` would give me an error saying that restore failed because it couldn't find `FSharp.Core 5.0.0`. I'd like to provide the exact error message, but that's not possible right now on my machine.

I tried the Repair in the installer to no avail.

`dotnet --version` returned the correct 5.0.0 RC version.

I uninstalled, but that finished instantly as soon as I hit Yes to UAC, I checked `c:\program files\dotnet\sdk`, and saw that it was still there, so I tried removing it manually.

That made it so that `dotnet --version` returns 3.1.402, but now when I go to try to reinstall RC1, it finishes as quickly as it uninstalled, without having actually installed it.

I've tried restarting several times throughout the process.

So now I have a borked zombie install which I can seem neither to install or to uninstall.

**Comment by pmout** [comment]
This work for me
Use:
netstat -ano | findstr ":9300.*LISTENING" 
To get "PID" from process that are using port 9300
Taskkill /PID "PID" /F
Run app again.

**Comment by dazinator** [comment]
I am having an issue where the following javascript file is now giving "404 not found" in the browser, after upgrading to RC1:

```
<script src="_framework/blazor.webassembly.js"></script>
```

Where can I find this file, is it sourced from some nuget package somewhere? How does it get incorporated into the project?

**Comment by captainsafia** [comment]
@dazinator Can you open an issue at https://github.com/dotnet/aspnetcore?

**Comment by dazinator** [comment]
@captainsafia - After an almost an hour looking around, it has suddenly starting working.. must have been some weird caching issue going on. I had tried disabling cache in the chrome network tab and that hadn't fixed it initially, but it seems to be working now! Sorry about that!

**Comment by BrettButcher** [comment]
Is there an approximate timeline for RC2, I would have thought fairly imminent? Holding on a breaking issue.
Thanks.

**Comment by rbhanda** [comment]
Hi @BrettButcher RC2 is expected to be released on 13th October 2020

**Comment by StefanOssendorf** [comment]
Hello,

where should I put false CA1416 warnings in projects with `net5.0-windows`?

We are using the `ManagementBaseObject` from the `System.Management` package v 5.0.0-rc.1.20451.14.
Project-file:
```csharp
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>net5.0-windows</TargetFramework>
        <GenerateDocumentationFile>true</GenerateDocumentationFile>
        <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
        <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
    </PropertyGroup>
    <ItemGroup>
        <SupportedPlatform Include="windows" />
    </ItemGroup>
    <ItemGroup>
        <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0-rc.1.20451.14" />
        <PackageReference Include="System.Management" Version="5.0.0-rc.1.20451.14" />
        <PackageReference Include="System.ServiceProcess.ServiceController" Version="5.0.0-rc.1.20451.14" />
        <PackageReference Include="Text.Analyzers" Version="2.6.4">
            <PrivateAssets>all</PrivateAssets>
            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
        </PackageReference>
        <PackageReference Include="System.Security.Cryptography.Xml" Version="5.0.0-rc.1.20451.14" />
    </ItemGroup>
</Project>
```
`Warning	CA1416	'ManagementBaseObject.this[string].set' is supported on 'windows'`.

**Comment by 4nTrax** [comment]
For everyone interested I done a simple workaround.

1.  Create a new ".bat" file and paste the following:
`for /f "tokens=5" %%a in ('netstat -aon ^| find ":9300" ^| find "LISTENING"') do taskkill /f /pid %%a`
This will find the PID of the process currently using port 9300 and kills it.

1.  With the file saved add the following lines to your Blazor `program.cs`, right at the start of the class:
```
            if (Debugger.IsAttached)
            {
                try
                {
                    ProcessStartInfo procInfo = new ProcessStartInfo();
                    procInfo.UseShellExecute = true;
                    procInfo.FileName = @"kill.bat";  //The file in that DIR.
                    procInfo.WorkingDirectory = @"C:\Users\user"; //The working DIR.
                    procInfo.Verb = "runas";
                    Process.Start(procInfo);  //Start that process.
                }
                catch (Exception)
                {}
            }
```

Pumped for RC2 ✌

**Comment by ariel-** [comment]
Since 5.0 RC 1, I noticed that projects with a project reference will copy the dll to both the output folder and to a `ref` subfolder.
Minimum repro:
```
dotnet new classlib -n MyLib
dotnet new console -n MyApp
dotnet new sln
```

Open the .sln, add both projects to solution, and add a project reference to MyApp (referencing MyLib). Build.
There is now a "MyLib.dll" under net5.0 folder and a "MyLib.dll" under net5.0\ref.

**Comment by rbhanda** [comment]
Closing in favor of https://github.com/dotnet/core/issues/5345
